Source/WebCore/dom/FullscreenManager.h

```diff
@@ -61,13 +61,15 @@ class FullscreenManager final : public CanMakeWeakPtr<FullscreenManager>, public
 
     // WHATWG Fullscreen API
     WEBCORE_EXPORT Element* fullscreenElement() const;
+    RefPtr<Element> protectedFullscreenElement() const { return fullscreenElement(); }
     WEBCORE_EXPORT bool isFullscreenEnabled() const;
     WEBCORE_EXPORT void exitFullscreen(RefPtr<DeferredPromise>&&);
 
     // Mozilla versions.
     bool isFullscreen() const { return m_fullscreenElement.get(); }
     bool isFullscreenKeyboardInputAllowed() const { return m_fullscreenElement.get() && m_areKeysEnabledInFullscreen; }
     Element* currentFullscreenElement() const { return m_fullscreenElement.get(); }
+    RefPtr<Element> protectedCurrentFullscreenElement() const { return currentFullscreenElement(); }
```

Comment:
We should really stop encouraging the use of `currentFullscreenElement` since it returns a bad value in some cases. I'd prefer if we'd just use `Ref { currentFullscreenElement() }` at the relevant call site until we replace it with `protectedFullscreenElement()`

Reply:
I disagree. We should not depart from our Safer C++ guidelines simply because we have an intention to re-architect some code later. Following the guidelines does not delay architecture changes; and meanwhile waiting for architecture changes does delay safety.